app/presenters/tree_builder_server_roles_common.rb

```diff
@@ -0,0 +1,7 @@
+module TreeBuilderServerRolesCommon
+  def x_build_single_node(object, pid, options)
+    options[:parent_kls]  = @sb[:parent_kls] if @sb.try(:parent_kls)
```

Comment:
@chessbyte now that this method is in a subclass, ``` @sb ``` will already be there, the code should be ok the way it is now.